src/libraries/Microsoft.PowerFx.Core/Binding/BinderUtils.cs

```diff
@@ -146,7 +146,17 @@ internal static bool TryGetBestOverload(CheckTypesContext context, IErrorContain
             TexlFunction matchingFuncWithCoercion = null;
             var matchingFuncWithCoercionReturnType = DType.Invalid;
             nodeToCoercedTypeMap = null;
-            Dictionary<TexlNode, DType> matchingFuncWithCoercionNodeToCoercedTypeMap = null;
+            Dictionary<TexlNode, DType> matchingFuncWithCoercionNodeToCoercedTypeMap = null;
+
+            // If a user-defined function exists as one of the possible overloads, always return that as it shadows existing built-in functions (provided it is not a restricted UDF name).
+            if (overloads.Any(overload => overload is UserDefinedFunction))
+            {
+                var udf = overloads.Where(overload => overload is UserDefinedFunction).First() as UserDefinedFunction;
+                var localWarnings = new LimitedSeverityErrorContainer(errors, DocumentErrorSeverity.Warning);
+                udf.CheckTypes(context, args, argTypes, localWarnings, out returnType, out nodeToCoercedTypeMap);
+                bestOverload = udf;
+                return true;

```

Comment:
>return true; [](http://example.com/codeflow?start=16&length=12)

should this always return true or should it be based on CheckTypes ?

also IIRC this path happens only when overloaded functions dont have lambda-arg/ ColumnNames-as-Id or something like that .

Can you also add some tests shadowing Filter , ShowColumns ?

Reply:
UDF doesn't support lambda params so a Filter test wouldn't be too interesting I think

Reply:
clarified offline , comment was to verify if the logic works shadowing functions with LambdaArg and ColumnNames as Identifiers